src/transformers/pipelines/zero_shot_classification.py

```diff
@@ -92,6 +110,20 @@ def _parse_and_tokenize(
         Parse arguments and tokenize only_first so that hypothesis (label) is not truncated
         """
         sequence_pairs = self._args_parser(sequences, candidate_labels, hypothesis_template)
+
+        if self.model.config.model_type == "t5":
+            sequence_pairs = [
+                prefix.format(premise=premise, hypothesis=hypothesis) for premise, hypothesis in sequence_pairs
+            ]
+
+            decoder_inputs = self.tokenizer(
+                ["<pad>"] * len(sequence_pairs),
```

Comment:
Why ?
This is probably wrong because we can't assume the pad_token is "<pad>" (self.tokenizer.pad_token is better).

Why would we force a pad prefix ?

Reply:
Agree. I have used the `"<pad>"` token because is the start token in the decoder of T5 model. But it is true that it should be the one defined in the model configuration.

```suggestion
                [self.model.config.decoder_start_token_id] * len(sequence_pairs),
```

We should ensure that `decoder_start_token_id` is defined, maybe with an assert like this:
```python
assert (
            self.model.config.decoder_start_token_id is not None
        ), "self.model.config.decoder_start_token_id has to be defined. In T5 it is usually set to the pad_token_id. See T5 docs for more information"
```

Reply:
Why do you need  start decoder token ? Why only one ?

Decoder don't necessarily have a starting token, do they ? BOS is pretty standard, but that's model dependant. (Could be a language token, for instance, or BOS + language token, no ?)

Reply:
I think there's `_prepare_decoder_input_ids_for_generation` that is used by `generate` to prepare the decoder tokens. (They can be forced too, but by default, each model will know how to set up generation)

Reply:
Okay